### PR TITLE
refactor summary sections

### PR DESCRIPTION
**Pull request by salgadopietrini**
---
name: Pull request
about: Create a pull request
title: ''
labels: ''
assignees: ''

---

## What type of PR is this?(check all applicable)
- [x] refactor
- [ ] feature
- [ ] bug fix
- [ ] documentation
- [ ] optimization
- [ ] other

### Related Issues

Issue #1066

### What this PR achieves

Currently, we have a common pattern where when we want to add a summary section (two columns), we create divs and add the classes that handle the grid layout. This PR encapsulates that logic into a single `<Summary />` component, which accepts two children (one for each column), and also implements it in places where the old logic was implemented:

- climb page 
- area page (through implementing it on `AreaPageContainer`)
- crag summary

### Screenshots, recordings
After refactor, should present no changes with current views:

- Area page, desktop
Screenshot 2024-04-17 at 18 01 57

- Climb page, desktop
Screenshot 2024-04-17 at 18 01 42

- Area page, mobile
Screenshot 2024-04-17 at 18 02 22

- Climb page, mobile 
Screenshot 2024-04-17 at 18 02 38

### Notes

- I saw that crag summary used the same pattern so I used the <Summary /> component there too. It should be pretty straightforward since the previous implementation was identical to the other section. However, I couldn't test changes on that section since I'm not very familiar yet so I don't how to reach that page.
- For each implementation, I extracted the component corresponding to the `left` and `right` columns into their own component. Let me know if this approach is ok or if you'd prefer that pass them directly to the props of `<Summary />`

### CODE REVIEW

**Comment by vnugent** [comment]
@salgadopietrini thanks for the PR.  I'll look into it this weekend.

File: src/app/(default)/components/ui/Summary.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  columns: {
+    left: ReactNode
+    right?: ReactNode
+  }
+}
+
+export const Summary: React.FC<Props> = ({ columns: { left, right } }) => (
+  <div className='area-summary'>
+    <div className='area-summary-left'>{left}</div>
+    {right !== undefined ? <div className='area-summary-right'>{right}</div> : null}
```
Comment (vnugent) [inline src/app/(default)/components/ui/Summary.tsx]: can you test for undefined with `!=` to catch both undefined and null?  Ex: 
```javascript
right != undefined ?
```

**Review by vnugent — CHANGES_REQUESTED** [review]

File: src/app/(default)/components/ui/Summary.tsx (hunk 1 above)
Comment (salgadopietrini) [reply src/app/(default)/components/ui/Summary.tsx]: yep, done!

**Comment by vnugent** [comment]
@allcontributors add @salgadopietrini for code

**Pull request merged**